controllers/cinder_controller.go

```diff
@@ -371,11 +448,15 @@ func (r *CinderReconciler) reconcileNormal(ctx context.Context, instance *cinder
 	// normal reconcile tasks
 	//
 
-	// TODO: create all sub-CRs, etc
-
 	// deploy cinder-api
 	cinderAPI, op, err := r.apiDeploymentCreateOrUpdate(instance)
 	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.DeploymentReadyCondition,
```

Comment:
I think we should create for the 4 cinder components (API,Scheduler, Backup, Volume) a condition type, like we have for the KeystoneService [1], then initialize them as unknown at the beginning [1] (like you already did with the default ones) and use the condition.mirror( componentType) [3] to get the latest condition status from the child object.

I think its enough for cinder volume to have a single one which shows the overall state of all cinder volumes.

[1] https://github.com/openstack-k8s-operators/keystone-operator/blob/master/api/v1beta1/conditions.go
[2] https://github.com/openstack-k8s-operators/keystone-operator/blob/master/controllers/keystoneservice_controller.go#L100
[3] https://github.com/openstack-k8s-operators/placement-operator/blob/master/controllers/placementapi_controller.go#L366-L371

Reply:
I had considered doing this initially.  I will give it a shot.